infer/tests/codetoanalyze/c/bo/test2.c

```diff
@@ -0,0 +1,18 @@
+#include<stdlib.h>
+
+void foo(int *arr, char *p, int i) {
+  int x = arr[0];
+  arr[x] = 1; /* BUG */
+
+  *(p + i) = 0; /* BUG */
+}
+
+int bar(int argc, char**argv){
+  int arr[10];
+  arr[0] = 100;
+  char* p = malloc(10);
+
+  foo(arr, p, 20);
+
+  return 0;
+}
```

Comment:
Adding some more tests would be really helpful for ensuring that we don't break existing functionality of inferbo. A few that I'd like to see

- Really simple tests reading directly from a bad index with no loop, like `int arr[1]; arr[1] = 2`
- Do..while loops
- Nested loops
- Loops with `break`/`continue`/`return`
- Nested loops with labeled `break`/`continue` (e.g., continue to outer loop head from inner loop)
- Infinite loops
- Unstructured loops formed using `goto`


It's ok if the analysis can't handle all of these features precisely, but it's really useful to have tests that make the capabilities and limitations of the analysis clear without actually reading the code for the analysis. Tests demonstrating a limitation are nice for pointing someone looking to get involved to a concrete improvement they can make.